src/Marten/Util/CommandExtensions.cs

```diff
@@ -59,7 +59,7 @@ public static NpgsqlCommand AsSproc(this NpgsqlCommand command)
 
         public static NpgsqlCommand WithJsonParameter(this NpgsqlCommand command, string name, string json)
         {
-            command.Parameters.Add("doc", NpgsqlDbType.Json).Value = json;
+            command.Parameters.Add("doc", NpgsqlDbType.Jsonb).Value = json;
```

Comment:
9.4 yells at you if you don't specify JSONB here and in SchemaBuilder (See above)